### PR TITLE
add more details to pipelined lock

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)
add more details to pipelined lock

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4958
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: pessimistic-transaction.md [hunk 1]
```diff
@@ -121,7 +121,13 @@ TiDB supports the following two isolation levels in the pessimistic transaction
...
+To reduce the overhead of locking, TiKV implements the pipelined locking process: when the data meets the requirements for locking, TiKV immediately notifies TiDB to execute subsequent requests and writes into the pessimistic lock asynchronously. This process reduces most latency and significantly improves the performance of pessimistic transactions. However, when TiKV has network isolation or nodes downtime, the asynchronous write into the pessimistic lock might fail, resulting in the following effects:
+
+* Other transactions that modify the same data cannot be blocked. If the application logic relies on locking or lock waiting mechanisms, it affects the correctness of the application logic.
+
+* Though there is a low probability that the transaction commit fails, it does not affect the correctness of the transactions.
```
Comment (cosven) [inline pessimistic-transaction.md:128]: The key point of this sentence is that **there is a low probability that the transaction commit fails**.

I'm not sure if we should change it to:
There is a low probability that the transaction commit fails, but it does not affect the correctness of the transactions.

**Review by cosven — COMMENTED** [review]
rest LGTM

cc @youjiali1995

File: pessimistic-transaction.md [hunk 2]
```diff
@@ -131,3 +137,11 @@ This feature is disabled by default. To enable it, modify the TiKV configuration
 [pessimistic-txn]
 pipelined = true
 ```
+
+If the TiKV cluster is v4.0.9 and above, you can also dynamically enable this feature by [modifying TiKV configuration online](/dynamic-config.md#modify-tikv-configuration-online):
```
Comment (TomShawn) [inline pessimistic-transaction.md:141]:
```suggestion
If the TiKV cluster is v4.0.9 and later, you can also dynamically enable this feature by [modifying TiKV configuration online](/dynamic-config.md#modify-tikv-configuration-online):
```

**Review by youjiali1995 — APPROVED** [review]

File: pessimistic-transaction.md [hunk 3]
```diff
@@ -121,7 +121,13 @@ TiDB supports the following two isolation levels in the pessimistic transaction
...
 
-To reduce the overhead of locking, TiKV implements the pipelined locking process: when the data meets the requirements for locking, TiKV immediately notifies TiDB to execute subsequent requests and writes into the pessimistic lock asynchronously. This process reduces most latency and significantly improves the performance of pessimistic transactions. However, there is a low probability that the asynchronous write into the pessimistic lock might fail, resulting in the commit failure of the pessimistic transaction.
+To reduce the overhead of locking, TiKV implements the pipelined locking process: when the data meets the requirements for locking, TiKV immediately notifies TiDB to execute subsequent requests and writes into the pessimistic lock asynchronously. This process reduces most latency and significantly improves the performance of pessimistic transactions. However, when network partition occurs in TiKV or a TiKV node is down, the asynchronous write into the pessimistic lock might fail and affects the following aspects:
```
Comment (TomShawn) [inline pessimistic-transaction.md:124]:
```suggestion
To reduce the overhead of locking, TiKV implements the pipelined locking process: when the data meets the requirements for locking, TiKV immediately notifies TiDB to execute subsequent requests and writes into the pessimistic lock asynchronously. This process reduces most latency and significantly improves the performance of pessimistic transactions. However, when network partition occurs in TiKV or a TiKV node is down, the asynchronous write into the pessimistic lock might fail and affect the following aspects:
```

File: pessimistic-transaction.md [hunk 4]
```diff
@@ -131,3 +137,11 @@ This feature is disabled by default. To enable it, modify the TiKV configuration
 [pessimistic-txn]
 pipelined = true
 ```
+
+If the TiKV cluster is v4.0.9 and later, you can also dynamically enable this feature by [modifying TiKV configuration online](/dynamic-config.md#modify-tikv-configuration-online):
```
Comment (TomShawn) [inline pessimistic-transaction.md:141]:
```suggestion
If the TiKV cluster is v4.0.9 or later, you can also dynamically enable this feature by [modifying TiKV configuration online](/dynamic-config.md#modify-tikv-configuration-online):
```

**Comment by TomShawn** [comment]
LGTM

**Review by TomShawn — APPROVED** [review]

**Pull request merged**